File: visual-dotnet/SauceLabs.Visual/Models/DiffingOption.cs
using System;

namespace SauceLabs.Visual.Models
{
    [Flags]
    public enum DiffingOption
    {
        None = 0,
        Content = 1 << 0,
        Dimensions = 1 << 1,
        Position = 1 << 2,
        Structure = 1 << 3,
        Style = 1 << 4,
        Visual = 1 << 5,
    }
}

File: visual-espresso/visual/src/main/java/com/saucelabs/visual/model/DiffingOption.java
package com.saucelabs.visual.model;

import com.saucelabs.visual.graphql.type.DiffingOptionsIn;

public enum DiffingOption {
    CONTENT,
    DIMENSIONS,
    POSITION,
    STRUCTURE,
    STYLE,
    VISUAL;

    public void apply(DiffingOptionsIn.Builder builder, boolean value) {
        if (this == CONTENT) {
            builder.content(value);
        }
        if (this == DIMENSIONS) {
            builder.dimensions(value);
        }
        if (this == POSITION) {
            builder.position(value);
        }
        if (this == STRUCTURE) {
            builder.structure(value);
        }
        if (this == STYLE) {
            builder.style(value);
        }
        if (this == VISUAL) {
            builder.visual(value);
        }
    }
}


